src/sentry/event_manager.py

```diff
@@ -2836,6 +2841,25 @@ def _detect_performance_problems(
         )
 
 
+def _update_user_reports_with_event_link(job: Job, group_info: GroupInfo) -> None:
```

Comment:
Maybe something like
```suggestion
def _update_user_reports_with_event_link(job: Job, group_info: GroupInfo) -> None:
'''Adds metrics and filters by missing group/environment to prevent duplicate shim'''
```

Reply:
this is going to change imminently so don't think there's a big point in adding a comment